ir/instr.cpp

```diff
@@ -680,6 +689,18 @@ unique_ptr<Instr> UnaryOp::dup(const string &suffix) const {
   return make_unique<UnaryOp>(getType(), getName() + suffix, *val, op, fmath);
 }
 
+TernaryOp::TernaryOp(Type &type, std::string &&name, Value &a, Value &b, Value &c, Op op,
+            FastMathFlags fmath)
+    : Instr(type, std::move(name)), a(&a), b(&b), c(&c), op(op), fmath(fmath) {
+  switch (op) {
+    case FShr:
+    case FShl:
+      assert(fmath.isNone());
+      break;
+    default:
```

Comment:
no default pls. That's a hazard waiting to explode in the future

Reply:
Fixed, changed to a case.